### DISCUSSION TITLE
How to manually change hyperparameters in Python? - Chemprop 2.1.0

### DISCUSSION THREAD

**Discussion by vdlmrc** [chemprop/chemprop#1114] [Q&A]
Hi Chemprop community,

Is there a way to manually change the hyperparameters in Python, without going through the whole hyperparameter optimization process in Chemprop 2.1.0.?

Let's say I want to reproduce a MPNN structure that I found in the literature and use that structure as a starting point, is it possible?

Many thanks!

Marc

**Comment by shihchengli** [comment]
Yes, it is doable. If you are an API-level user, we have an [example notebook](https://chemprop.readthedocs.io/en/main/training.html) to guide users on how to run Chemprop in a notebook, where you can customize the message passing block and predictor block with the hyperparameters you want. Specifically, you can read our module-level tutorials here ([Link1](https://chemprop.readthedocs.io/en/main/tutorial/python/models/message_passing.html#Customization), [Link2](https://chemprop.readthedocs.io/en/main/tutorial/python/models/predictor.html#Customization)). If you are a CLI-level user, you can pass the hyperparameters you want with the corresponding flags (see [here](https://chemprop.readthedocs.io/en/main/tutorial/cli/train.html#hyperparameters)).

**Comment by vdlmrc** [comment]
Many thanks!
